File: other/Solution118.java
import java.util.List;
import java.util.LinkedList;

class Solution118 {
    public List<List<Integer>> generate(int numRows) {
    	if(numRows == 0){
    		List<List<Integer>> result = new LinkedList<>();
    		return result;
    	}
    	if(numRows == 1){
    		List<List<Integer>> result = new LinkedList<>();
    		List<Integer> list = new LinkedList<>();
    		list.add(1);
    		result.add(list);
    		return result;
    	}
    	List<List<Integer>> list = generate(numRows - 1);
    	int index = 0;
    	List<Integer> last = list.get(list.size() - 1);
    	List<Integer> cur = new LinkedList<>();
    	cur.add(1);
    	while(index + 1 < last.size()){
    		cur.add(last.get(index) + last.get(index + 1));
    		index += 1;
    	}
    	cur.add(1);
    	list.add(cur);
    	return list;
    }

    public static void main(String[] args){
    	Solution118 s = new Solution118();
    	List<List<Integer>> result = s.generate(5);
    	for(List<Integer> ele: result){
    		System.out.println(ele);
    	}
    }
}

File: C++/Solution118.cpp
#include<iostream>
#include<vector>
#include<algorithm>
#include<unordered_map>
#include<unordered_set>
#include<queue>
using namespace std;

typedef long long ll;
typedef pair<int, int> pii;


class Solution118 {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> ans;
        ans.push_back({1});
        while(numRows > 1){
            auto& last = ans.back();
            vector<int> cur;
            cur.push_back(1);
            for(int i = 0; i + 1 < last.size(); i++){
                cur.push_back(last[i] + last[i + 1]);
            }
            cur.push_back(1);
            ans.push_back(cur);
            numRows --;
        }
        return ans;
    }
};